linalg/densemat.cpp

```diff
@@ -436,6 +455,43 @@ double DenseMatrix::Det() const
             d[3] * (d[2] * d[7] - d[1] * d[8]) +
             d[6] * (d[1] * d[5] - d[2] * d[4]);
       }
+      case 4:
+      {
+         const double *d = data;
+         return
+            d[ 0] * (d[ 5] * (d[10] * d[15] - d[11] * d[14]) -
```

Comment:
I added a Det() method to the LUFactors class which computes the determinant assuming the factorization has been computed first.  This had to be done inside of LUFactors because it needs access to the ipiv_base variable in order to correctly interpret the permutation information stored in ipiv.

I ran a series of tests comparing the previous DenseMatrix recursive implementation with the new LUFactors::Det() implementation.  These tests compared the computed determinants and timings for thousands of random matrices.  The values agreed to at least 6 decimal places (I didn't check for equality just that the difference was sufficiently small).  The timings showed that the hard-coded cases for 1x1, 2x2, 3x3, and 4x4 matrices were quicker than LUFactors::Det() and the recursive algorithm was slower for 5x5 or larger matrices (I'm glad Veselin pointed this out because I would not have thought of this). 

I then modified the DenseMatrix::Det() method to use LUFactors::Det() whenever the matrix size is larger than 4x4.